### PR TITLE
Incorporate name detection into SmartTextVectorizer

### PR DESCRIPTION
**Pull request by Jauntbox**
**Related issues**
Re-opening of https://github.com/salesforce/TransmogrifAI/pull/456 on branch directly on TransmogrifAI, please see that PR for historical discussion (I'll close once this one gets merged in).

**Describe the proposed solution**
Adds parameters for ignoring fields identified as personal names to SmartTextVectorizer

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### CODE REVIEW

**Comment by Jauntbox** [comment]
Right now, this is just compiling and passing tests so far. I still have to give it a pass to clean things up before it's ready for a real review

File: core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala [hunk 1]
```diff
@@ -84,14 +90,30 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
...
+    } else {
+      val nameDetectMapFun = makeMapFunction(dataset.sparkSession)
+      dataset.map(_.map(input => (TextStats.computeTextStats(input, shouldCleanText, shouldTokenizeForLengths, maxCard),
+          nameDetectMapFun(input))
+      ).toArray).reduce(_ + _).unzip
+    }
+    // val valueStats: Dataset[Array[TextStats]] = dataset.map(
+    //   _.map(TextStats.computeTextStats(_, shouldCleanText, shouldTokenizeForLengths, maxCard)).toArray
+    // )
+    // val aggregatedStats: Array[TextStats] = valueStats.reduce(_ + _)
```
Comment (leahmcguire) [inline core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala:111]: remove commented lines

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/SmartTextMapVectorizerTest.scala [hunk 2]
```diff
@@ -31,14 +31,26 @@
...
+>>>>>>> 4d461814d1791ef7e6dd14b262cf6a8a303ffcde
 import com.salesforce.op.test.{OpEstimatorSpec, TestFeatureBuilder}
 import com.salesforce.op.testkit.RandomText
 import com.salesforce.op.utils.spark.RichDataset._
+<<<<<<< HEAD
+import com.salesforce.op.utils.stages.{NameDetectUtils, SensitiveFeatureMode}
+import org.apache.log4j.Level
+=======
 import com.salesforce.op.utils.spark.{OpVectorColumnMetadata, OpVectorMetadata}
+>>>>>>> 4d461814d1791ef7e6dd14b262cf6a8a303ffcde
```
Comment (leahmcguire) [inline core/src/test/scala/com/salesforce/op/stages/impl/feature/SmartTextMapVectorizerTest.scala:53]: please fix merge conflict

**Review by leahmcguire — COMMENTED** [review]
Just minor stuff - otherwise looks good

**Review by leahmcguire — APPROVED** [review]

**Pull request merged**